src/libraries/System.ComponentModel.Primitives/src/System/ComponentModel/InvalidEnumArgumentException.cs

```diff
@@ -43,12 +43,13 @@ public InvalidEnumArgumentException(string? message, Exception? innerException)
         /// class with a message generated from the argument, invalid value, and
         /// enumeration class.
         /// </summary>
-        public InvalidEnumArgumentException(string? argumentName, int invalidValue, Type enumClass!!)
+        public InvalidEnumArgumentException(string? argumentName, int invalidValue, Type enumClass)
             : base(SR.Format(SR.InvalidEnumArgument,
                                 argumentName,
                                 invalidValue,
-                                enumClass.Name), argumentName)
+                                enumClass?.Name), argumentName)
         {
+            ArgumentNullException.ThrowIfNull(enumClass);
         }
```

Comment:
> I'm curious why you chose not to throw from inside the call to base here (as was done in other occurrences of a similar scenario)

`throw new ArgumentNullException(nameof(enumClass))` results in larger code.  I only use it in this(...)/base(...) when I'm forced to in order to avoid null refs.  It's one of the nice things that `!!` provided, having the smaller code size of `ArgumentNullException.ThrowIfNull` but also being able to do that before the delegation.  We're losing that along with `!!`.

Reply:
Got it; thanks.